lib/helpers.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# Helper functions
+
+# Extract item or items matching .name, including deduplication
+# $1: item name, prepended or not with deduplication mark
+array_from_name() {
+  item_name="$(echo "$1" | sed "s/$DEDUP_MARK //")"
+  echo "$ITEMS" | jq -r ". | map(select((.name == \"$item_name\") and (.type == $TYPE_LOGIN)))"
+}
+
+# Extract item matching .id
+# $1: string starting with ".id:"
+array_from_id() {
+  item_id="$(echo "$1" | cut -d ':' -f 1)"
```

Comment:
You're already filtering the id before calling this function (which imo is the cleaner way). So I would remove this `cut -d ':' -f 1` here and only allow the pure id to be passed

Reply:
Yeah I was getting a bug so i wrote it before calling the function... probably didn't wrap the variable in `"` or something...